src/app/workflow/workflow-file-editor/workflow-file-editor.component.ts

```diff
@@ -21,8 +21,11 @@ export class WorkflowFileEditorComponent extends FileEditing {
   @Input() set selectedVersion(value: WorkflowVersion) {
     this._selectedVersion = value;
     if (value != null) {
+      this.clearSourceFiles();
```

Comment:
That said, this seems redundant. 
If in both branches you use `this.clearSourceFiles()` just take it out of the if completely and have it always execute afterwards.